File: db/migrate.py
import asyncio
import asyncpg
from db.schema import SCHEMA_SQL
from config.settings import settings
from utils.logger import get_logger

log = get_logger("Migrate")


async def run():
    conn = await asyncpg.connect(settings.database_url)
    log.info("Running migrations...")
    await conn.execute(SCHEMA_SQL)
    await conn.close()
    log.info("Migrations complete ✅")


if __name__ == "__main__":
    asyncio.run(run())

File: db/migrate.js
import pg from "pg";
import config from "../config/index.js";
import { schema } from "./schema.js";
import { createLogger } from "../utils/logger.js";

const log = createLogger("Migrate");

async function migrate() {
  const client = new pg.Client({ connectionString: config.db.url });
  await client.connect();
  log.info("Running migrations...");
  await client.query(schema);
  log.info("Migrations complete ✅");
  await client.end();
}

migrate().catch((err) => {
  console.error("Migration failed:", err);
  process.exit(1);
});
